rs/execution_environment/fuzz/src/lib.rs

```diff
@@ -85,13 +85,27 @@ fn syscall_monitor<F>(name: &str, sandbox: F)
 where
     F: Fn(),
 {
+    // The number of sandbox threads must be known in advance because:
+    //   1. We need to call `ptrace::attach` immediately after the thread is spawned.
+    //   2. Detached threads interfere with libfuzzer's cleanup process.
```

Comment:
Not sure I follow part 2 - you're saying `syscall_moniter` gets run in it's own thread and we need to ensure it exits on completion? It seems like that might be possible without hard-coding the number of threads.

Reply:
The `syscall_monitor` is the parent process and it spawns a tracer in a thread for each thread of the sandbox (which is invoked in the child process). After all tracer threads are spawned, the parent process must panic if a the tracer thread panics for proper cleanup.

We achieve this by counting expected tracer threads and `join().unwrap()` on them. Do you see a alternative way around this?

Reply:
Ah, I missed the sandbox exits as well in this case. I have updated accordingly.